src/g_array.c

```diff
@@ -475,90 +475,64 @@ void garray_arraydialog(t_garray *x, t_symbol *name, t_floatarg fsize,
 }
 
 /* jsarlo { */
-static int garray_arrayviewlist_setpage(t_symbol*arrayname, t_array*a, int page, int pagesize)
+static void garray_arrayviewlist_fillpage(t_garray *x,
+                                   t_float fPage,
+                                   t_float fTopItem)
 {
+    int i, size=0, topItem=(int)fTopItem;
+    int pagesize=ARRAYPAGESIZE, page=(int)fPage, maxpage;
+    t_word *data=0;
 
-        /* make szre the requested page is within range */
-    int maxpage = (a->a_n - 1) / pagesize;
-    if(page < 0)
-        page = 0;
+    if(!garray_getfloatwords(x, &size, &data)) {
+        pd_error(x, "error in %s()", __FUNCTION__);
```

Comment:
Small nitpick: I think that `bug()` would be more appropriate here, as the function should never fail. Same in `garray_arrayviewlist_new`.

Reply:
probaby.
but then, `pd_error()` allows us to trace the error to the problematic array, whereas `bug()` does not (but gives us an extra 'consistency check failed').

which one should be preferred?

(`error()` has been used in this place since the original implementation in 2004 :stuck_out_tongue: )